### ISSUE TITLE
An extra unenrollment ping is registered when unenrolling from the study while in treatmentDomain branch 

### ISSUE THREAD

**Issue by SoftVision-CosminMuntean** [mozilla/fingerprinting-blocking-methods-study-addon#9]
[Affected Platforms]:
- All Windows
- All Mac
- All Linux

[Affected Versions]:
- Firefox 69.0 Unbranded build
- Firefox 70.0.1 Unbranded build
- Firefox 71 Unbranded build

[Prerequisites]:
- Have a Firefox 69, 70 or 71 Unbranded build. 
- Have access to the stage delivery console.
- Have an active branched-addon recipe for the Fingerprinting Implementation Retention Study V1. You can use my [created recipe](https://delivery-console.stage.mozaws.net/recipe/882).
- Have a user.js file that contains all the prefs needed to be targeted by the Normandy recipe. You can use my created [user.js](https://drive.google.com/open?id=12j8J7WTMU-ZvHjcUhcutB09Qau_GQ24_).

[Steps to reproduce]:
1. Open the browser with the profile from prerequisites.
2. In the profile directory copy the user.js file.
3. Restart the browser.
4. Navigate to "about:studies" page and click the "Remove" button of the Fingerprinting study.
5. Navigate to "about:telemetry#events-tab" page and search for an unenroll event.

[Expected result]:
- One unenroll telemetry event is sent and its reason is "individual-opt-out".

[Actual result]:
- Two unenroll telemetry events are sent. 
	One is registered for "individual-opt-out".
	The other one is registered for "2: User has turned fingerprinting protections off.".
	
[Notes]:
- The issue is also reproducible when the Normandy recipe is disabled. The following unenroll events are sent: 
![disable recip](https://user-images.githubusercontent.com/19226534/69881454-c45d3c00-12d5-11ea-8355-22ec841a1faa.png)

- The issue is also reproducible when the "app.shield.optoutstudies.enabled" pref is set to false. The following unenroll events are sent: 
![general opt-ou](https://user-images.githubusercontent.com/19226534/69881446-be675b00-12d5-11ea-9aae-d245804e46a7.png)

- The issue is not reproducible if the add-on is removed from the "Experiments" folder. Only the unenroll event with the "uninstalled-sideload" reason is registered.
- The "2: User has turned fingerprinting protections off." event is correctly registered if manually flit the FP-PREF.
- The unenroll event are correctly registered for the Control and treatmentScript branches in all cases.
- Here is a screen recording with the issue: [link](https://drive.google.com/open?id=13P4yxeabjr4Fmv-HACokiA3noiHwC8jF).

**Comment by birdsarah** [comment]
@SoftVision-CosminMuntean thank you very much for the video it was extremely helpful.

@motin can you help me figure this out? Here's code where I trigger the unenroll on treatmentDomain: https://github.com/mozilla/fingerprinting-blocking-methods-study-addon/blob/master/src/treatmentDomain/studyController.js#L45

I'm not 100% sure what's happening, but if I had to guess:
* the user unenrolls
* the unenrollment causes the pref to flip back to off - resetting the state to pre-experiment state
* this triggers a second unenrollment event

I hadn't coded for this. What have you done about this kind of thing before @motin? Can you suggest a good pattern?

**Comment by SoftVision-CosminMuntean** [comment]
I have verified this issue using the latest builds (0.4.0) and the issue is no longer reproducible. Only one specific unenroll telemetry event is correctly sent. 
I have verified the issue using all the unenroll methods:
- Removing the study from "about:studies" page.
- Manually change pref from "about:config" page.
- Disabled recipe from delivery console
- Removing the extension from the "Extensions" folder.
- Setting the "app.shield.optoutstudies.enabled" pref to "true".
